### ISSUE TITLE
Support for granting a permission to a canonical user id

### ISSUE THREAD

**Issue by yohei1126** [fluent/fluent-plugin-s3#212]
I have a bucket policy as follows to reject putObject request without grant-full-control for bucket owner's canonical user id for some reasons. The plugin currently supports `acl` but  it would be very helpful if it also supports `grant_full_control`.

```
{
    "Sid": "DenyPutObjectWithoutGrantFullControl",
    "Effect": "Deny",
    "Principal": "*",
    "Action": "s3:PutObject",
    "Resource": "arn:aws:s3:::my-bucket/*",
    "Condition": {
        "StringNotEquals": {
            "s3:x-amz-grant-full-control": "id=0123456789"
        }
    }
}
```

Please refer to [AWS documentation](http://docs.aws.amazon.com/AmazonS3/latest/dev/example-walkthroughs-managing-access-example3.html) for more details about this option.
For AWS CLI, this option is uses as follows.

```
$ aws s3api put-object-acl --bucket examplebucket --key HappyFace.jpg \
   --grant-full-control id="AccountA-CanonicalUserID" --profile AccountBadmin
```

**Comment by yohei1126** [comment]
fix for v0.12 is merged now. I will also send pull request to master later.

**Comment by yohei1126** [comment]
Confirmed that we can use grant-full-control on fluent-plugin-s3 ver. 0.8.7.
https://rubygems.org/gems/fluent-plugin-s3/versions/0.8.7
